src/main/scala/io/iohk/ethereum/nodebuilder/tooling/PeriodicConsistencyCheck.scala

```diff
@@ -0,0 +1,52 @@
+package io.iohk.ethereum.nodebuilder.tooling
+
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.scaladsl.TimerScheduler
+
+import scala.concurrent.duration.DurationInt
+
+import io.iohk.ethereum.db.storage.AppStateStorage
+import io.iohk.ethereum.db.storage.BlockHeadersStorage
+import io.iohk.ethereum.db.storage.BlockNumberMappingStorage
+import io.iohk.ethereum.nodebuilder.tooling.PeriodicConsistencyCheck.ConsistencyCheck
+import io.iohk.ethereum.utils.Logger
+
+object PeriodicConsistencyCheck {
+  def start(
+      appStateStorage: AppStateStorage,
+      blockNumberMappingStorage: BlockNumberMappingStorage,
+      blockHeadersStorage: BlockHeadersStorage,
+      shutdown: () => Unit
+  ): Behavior[ConsistencyCheck] =
+    Behaviors.withTimers { timers =>
+      timers.startSingleTimer(Tick, 10.minutes)
+      PeriodicConsistencyCheck(timers, appStateStorage, blockNumberMappingStorage, blockHeadersStorage, shutdown)
+        .check()
+    }
+
+  sealed trait ConsistencyCheck extends Product with Serializable
+  case object Tick extends ConsistencyCheck
+}
+
+case class PeriodicConsistencyCheck(
+    timers: TimerScheduler[ConsistencyCheck],
+    appStateStorage: AppStateStorage,
+    blockNumberMappingStorage: BlockNumberMappingStorage,
+    blockHeadersStorage: BlockHeadersStorage,
+    shutdown: () => Unit
+) extends Logger {
+  import PeriodicConsistencyCheck._
+
+  def check(): Behavior[ConsistencyCheck] = Behaviors.receiveMessage { case Tick =>
+    log.debug("Running a storageConsistency check")
+    StorageConsistencyChecker.checkStorageConsistency(
+      appStateStorage.getBestBlockNumber(),
+      blockNumberMappingStorage,
+      blockHeadersStorage,
+      shutdown
+    )(log)
+    timers.startSingleTimer(Tick, 5.seconds)
```

Comment:
It's a bit confusing. Are we doing the check every 10 minutes (https://github.com/input-output-hk/mantis/pull/1070/files#diff-a8c02d6306b1c231f0efde1b3f46fa9cc6659899bfa882c317479432df489a7aR23) or 5 seconds? 
It's probably a good idea to make this configurable, unless it's just a temporary measure

Reply:
good catch, this should be a method